wgpu-core/src/device.rs

```diff
@@ -1203,7 +1230,7 @@ impl<F: IdentityFilter<BindGroupLayoutId>> Global<F> {
 
         let layout = binding_model::BindGroupLayout {
             raw,
-            bindings: bindings.to_vec(),
+            bindings: bindings.iter().cloned().map(|b| (b.binding, b)).collect(),
```

Comment:
This is the allocation that we can move up to simplify the comparison and also make it more correct. But we may end up not using it.